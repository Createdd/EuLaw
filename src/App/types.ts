interface AppStateType {
  currentUser: string;
}

export {
  AppStateType,
};