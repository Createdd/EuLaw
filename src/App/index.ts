export {App as default} from './App';
export { AppStateType } from './types';