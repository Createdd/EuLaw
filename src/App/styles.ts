import styled, { injectGlobal } from 'styled-components';

/* tslint:disable */
injectGlobal`
  body {
    margin: 0;
    padding: 0;
    font-family: sans-serif;
  }
`;
/* tslint:enable */

const AppStyle = styled.div`
  text-align: center;

  & img {
  animation: App-logo-spin infinite 20s linear;
  height: 2rem;
  }

  & h1 {
    font-size: 1.5em;
  }

  @keyframes App-logo-spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
  }
`;

const HeaderStyle = styled.header`
  background-color: #222;
  height: 5rem;
  padding: 1rem;
  color: white;
`;

export {
  AppStyle,
  HeaderStyle
};