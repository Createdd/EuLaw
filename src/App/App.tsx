import * as React from 'react';
var nlp = require('compromise');

import { AppStateType } from './types';
import { AppStyle, HeaderStyle } from './styles';

import ExampleCase from '../data/exampleData/exampleCase';
import Chart, { ChartDataType } from '../components/Chart';
import TextField from '../components/TextField';
import DecisionResults from '../components/DecisionResults';
import TextAnalytics from '../components/TextAnalytics';

const logo = require('../logo.svg');
const data: ChartDataType = [
  { name: 'Page A', uv: 4000, pv: 2400, amt: 2400 },
  { name: 'Page B', uv: 3000, pv: 1398, amt: 2210 },
  { name: 'Page C', uv: 2000, pv: 9800, amt: 2290 },
  { name: 'Page D', uv: 2780, pv: 3908, amt: 2000 },
  { name: 'Page E', uv: 1890, pv: 4800, amt: 2181 },
  { name: 'Page F', uv: 2390, pv: 3800, amt: 2500 },
  { name: 'Page G', uv: 3490, pv: 4300, amt: 2100 }
];

const results = '1000';

const doc = nlp(ExampleCase.text);
console.log(
  doc
    .topics()
    .slice(0, 4)
    .out('frequency')
);

/* tslint:disable no-any */
class App extends React.Component<any, AppStateType> {
  constructor(props: any) {
    super(props);
    this.state = {
      currentCase: 'Paste example case here'
    };
  }

  getValueHandler = (text: string) => {
    console.error(text);
    this.setState({
      currentCase: text
    });
  }

  render(): JSX.Element {
    return (
      <AppStyle>
        <HeaderStyle>
          <img src={logo} alt="logo" />
          <h1>Welcome to EU Fundamentals Crunch</h1>
        </HeaderStyle>
        <h2>Paste example case here</h2>
        <TextField getValue={this.getValueHandler} />
        <TextAnalytics text={this.state.currentCase}/>
        <h2>Example Case:</h2>â€š
        <p>{ExampleCase.text}</p>
        <DecisionResults results={results} />
        <Chart data={data} />
      </AppStyle>
    );
  }
}

export { App };
