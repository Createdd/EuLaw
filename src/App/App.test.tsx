import * as React from 'react';
import * as ReactDOM from 'react-dom';
import * as TestRenderer from 'react-test-renderer';
import { App } from './App';

it('renders the component', () => {
  const result = TestRenderer.create(<App />).toJSON();
  expect(result).toMatchSnapshot();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});
