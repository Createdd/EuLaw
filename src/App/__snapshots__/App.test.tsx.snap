// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="sc-bdVaJa dukJkZ"
>
  <header
    className="sc-bwzfXH kJcZVs"
  >
    <img
      alt="logo"
      src="logo.svg"
    />
    <h1>
      Welcome to EU Fundamentals Crunch
    </h1>
  </header>
  <h2>
    Paste example case here
  </h2>
  <form
    className="sc-bxivhb jsuVEO"
    onSubmit={[Function]}
  >
    <textarea
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Press ENTER to send!"
      value=""
    />
  </form>
  <form>
    <div
      className="sc-ifAKCX jtdfU"
    >
      <label>
        <input
          checked={true}
          disabled={true}
          name="isMemberState"
          onChange={[Function]}
          type="checkbox"
        />
        is a member state
      </label>
    </div>
  </form>
  <div
    className="sc-htpNat gcoSfS"
  >
    <div
      className="recharts-wrapper"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onTouchMove={[Function]}
      style={
        Object {
          "cursor": "default",
          "height": 300,
          "position": "relative",
          "width": 600,
        }
      }
    >
      <svg
        className="recharts-surface"
        height={300}
        style={undefined}
        version="1.1"
        viewBox="0 0 600 300"
        width={600}
      >
        <g
          className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
        >
          <line
            className="recharts-cartesian-axis-line"
            fill="none"
            height={30}
            stroke="#666"
            width={490}
            x={80}
            x1={80}
            x2={570}
            y={265}
            y1={265}
            y2={265}
          />
          <g
            className="recharts-cartesian-axis-ticks"
          >
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                stroke="#666"
                width={490}
                x={80}
                x1={80}
                x2={80}
                y={265}
                y1={271}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={30}
                stroke="none"
                textAnchor="middle"
                width={490}
                x={80}
                y={273}
              >
                <tspan
                  dy="0.71em"
                  x={80}
                >
                  Page A
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                stroke="#666"
                width={490}
                x={80}
                x1={161.66666666666669}
                x2={161.66666666666669}
                y={265}
                y1={271}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={30}
                stroke="none"
                textAnchor="middle"
                width={490}
                x={161.66666666666669}
                y={273}
              >
                <tspan
                  dy="0.71em"
                  x={161.66666666666669}
                >
                  Page B
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                stroke="#666"
                width={490}
                x={80}
                x1={243.33333333333334}
                x2={243.33333333333334}
                y={265}
                y1={271}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={30}
                stroke="none"
                textAnchor="middle"
                width={490}
                x={243.33333333333334}
                y={273}
              >
                <tspan
                  dy="0.71em"
                  x={243.33333333333334}
                >
                  Page C
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                stroke="#666"
                width={490}
                x={80}
                x1={325}
                x2={325}
                y={265}
                y1={271}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={30}
                stroke="none"
                textAnchor="middle"
                width={490}
                x={325}
                y={273}
              >
                <tspan
                  dy="0.71em"
                  x={325}
                >
                  Page D
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                stroke="#666"
                width={490}
                x={80}
                x1={406.6666666666667}
                x2={406.6666666666667}
                y={265}
                y1={271}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={30}
                stroke="none"
                textAnchor="middle"
                width={490}
                x={406.6666666666667}
                y={273}
              >
                <tspan
                  dy="0.71em"
                  x={406.6666666666667}
                >
                  Page E
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                stroke="#666"
                width={490}
                x={80}
                x1={488.33333333333337}
                x2={488.33333333333337}
                y={265}
                y1={271}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={30}
                stroke="none"
                textAnchor="middle"
                width={490}
                x={488.33333333333337}
                y={273}
              >
                <tspan
                  dy="0.71em"
                  x={488.33333333333337}
                >
                  Page F
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={30}
                stroke="#666"
                width={490}
                x={80}
                x1={570}
                x2={570}
                y={265}
                y1={271}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={30}
                stroke="none"
                textAnchor="middle"
                width={490}
                x={570}
                y={273}
              >
                <tspan
                  dy="0.71em"
                  x={570}
                >
                  Page G
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g
          className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
        >
          <line
            className="recharts-cartesian-axis-line"
            fill="none"
            height={260}
            stroke="#666"
            width={60}
            x={20}
            x1={80}
            x2={80}
            y={5}
            y1={5}
            y2={265}
          />
          <g
            className="recharts-cartesian-axis-ticks"
          >
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={260}
                stroke="#666"
                width={60}
                x={20}
                x1={74}
                x2={80}
                y={5}
                y1={265}
                y2={265}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={260}
                stroke="none"
                textAnchor="end"
                width={60}
                x={72}
                y={265}
              >
                <tspan
                  dy="0.355em"
                  x={72}
                >
                  0
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={260}
                stroke="#666"
                width={60}
                x={20}
                x1={74}
                x2={80}
                y={5}
                y1={200}
                y2={200}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={260}
                stroke="none"
                textAnchor="end"
                width={60}
                x={72}
                y={200}
              >
                <tspan
                  dy="0.355em"
                  x={72}
                >
                  2500
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={260}
                stroke="#666"
                width={60}
                x={20}
                x1={74}
                x2={80}
                y={5}
                y1={135}
                y2={135}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={260}
                stroke="none"
                textAnchor="end"
                width={60}
                x={72}
                y={135}
              >
                <tspan
                  dy="0.355em"
                  x={72}
                >
                  5000
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={260}
                stroke="#666"
                width={60}
                x={20}
                x1={74}
                x2={80}
                y={5}
                y1={70}
                y2={70}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={260}
                stroke="none"
                textAnchor="end"
                width={60}
                x={72}
                y={70}
              >
                <tspan
                  dy="0.355em"
                  x={72}
                >
                  7500
                </tspan>
              </text>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis-tick"
            >
              <line
                className="recharts-cartesian-axis-tick-line"
                fill="none"
                height={260}
                stroke="#666"
                width={60}
                x={20}
                x1={74}
                x2={80}
                y={5}
                y1={5}
                y2={5}
              />
              <text
                className="recharts-text recharts-cartesian-axis-tick-value"
                fill="#666"
                height={260}
                stroke="none"
                textAnchor="end"
                width={60}
                x={72}
                y={5}
              >
                <tspan
                  dy="0.355em"
                  x={72}
                >
                  10000
                </tspan>
              </text>
            </g>
          </g>
        </g>
        <g
          className="recharts-cartesian-grid"
        >
          <g
            className="recharts-cartesian-grid-horizontal"
          >
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={80}
              x2={570}
              y={5}
              y1={265}
              y2={265}
            />
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={80}
              x2={570}
              y={5}
              y1={200}
              y2={200}
            />
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={80}
              x2={570}
              y={5}
              y1={135}
              y2={135}
            />
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={80}
              x2={570}
              y={5}
              y1={70}
              y2={70}
            />
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={80}
              x2={570}
              y={5}
              y1={5}
              y2={5}
            />
          </g>
          <g
            className="recharts-cartesian-grid-vertical"
          >
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={80}
              x2={80}
              y={5}
              y1={5}
              y2={265}
            />
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={161.66666666666669}
              x2={161.66666666666669}
              y={5}
              y1={5}
              y2={265}
            />
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={243.33333333333334}
              x2={243.33333333333334}
              y={5}
              y1={5}
              y2={265}
            />
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={325}
              x2={325}
              y={5}
              y1={5}
              y2={265}
            />
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={406.6666666666667}
              x2={406.6666666666667}
              y={5}
              y1={5}
              y2={265}
            />
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={488.33333333333337}
              x2={488.33333333333337}
              y={5}
              y1={5}
              y2={265}
            />
            <line
              fill="none"
              height={260}
              stroke="#ccc"
              strokeDasharray="3 3"
              width={490}
              x={80}
              x1={570}
              x2={570}
              y={5}
              y1={5}
              y2={265}
            />
          </g>
        </g>
        <g
          className="recharts-layer recharts-line"
        >
          <path
            className="recharts-curve recharts-line-curve"
            clipPath={null}
            d="M80,202.6C107.22222222222223,215.62599999999998,134.44444444444446,228.652,161.66666666666669,228.652C188.8888888888889,228.652,216.11111111111111,10.200000000000017,243.33333333333334,10.200000000000017C270.55555555555554,10.200000000000017,297.77777777777777,163.392,325,163.392C352.22222222222223,163.392,379.44444444444446,140.2,406.6666666666667,140.2C433.8888888888889,140.2,461.11111111111114,166.2,488.33333333333337,166.2C515.5555555555555,166.2,542.7777777777778,159.7,570,153.2"
            fill="none"
            height={260}
            stroke="#8884d8"
            strokeDasharray="0px 0px"
            strokeWidth={1}
            width={490}
          />
        </g>
        <g
          className="recharts-layer recharts-line"
        >
          <path
            className="recharts-curve recharts-line-curve"
            clipPath={null}
            d="M80,161C107.22222222222223,169.66666666666666,134.44444444444446,178.33333333333334,161.66666666666669,187C188.8888888888889,195.66666666666666,216.11111111111111,213,243.33333333333334,213C270.55555555555554,213,297.77777777777777,192.72,325,192.72C352.22222222222223,192.72,379.44444444444446,215.86,406.6666666666667,215.86C433.8888888888889,215.86,461.11111111111114,209.79333333333335,488.33333333333337,202.86C515.5555555555555,195.92666666666668,542.7777777777778,185.09333333333333,570,174.26"
            fill="none"
            height={260}
            stroke="#82ca9d"
            strokeDasharray="0px 0px"
            strokeWidth={1}
            width={490}
          />
        </g>
      </svg>
      <div
        className="recharts-legend-wrapper"
        style={
          Object {
            "bottom": 5,
            "height": "auto",
            "left": 20,
            "position": "absolute",
            "width": 550,
          }
        }
      >
        <ul
          className="recharts-default-legend"
          style={
            Object {
              "margin": 0,
              "padding": 0,
              "textAlign": "center",
            }
          }
        >
          <li
            className="recharts-legend-item legend-item-0"
            style={
              Object {
                "display": "inline-block",
                "marginRight": 10,
              }
            }
          >
            <svg
              className="recharts-surface"
              height={14}
              style={
                Object {
                  "display": "inline-block",
                  "marginRight": 4,
                  "verticalAlign": "middle",
                }
              }
              version="1.1"
              viewBox="0 0 32 32"
              width={14}
            >
              <path
                className="recharts-legend-icon"
                d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                fill="none"
                stroke="#8884d8"
                strokeWidth={4}
              />
            </svg>
            <span
              className="recharts-legend-item-text"
            >
              pv
            </span>
          </li>
          <li
            className="recharts-legend-item legend-item-1"
            style={
              Object {
                "display": "inline-block",
                "marginRight": 10,
              }
            }
          >
            <svg
              className="recharts-surface"
              height={14}
              style={
                Object {
                  "display": "inline-block",
                  "marginRight": 4,
                  "verticalAlign": "middle",
                }
              }
              version="1.1"
              viewBox="0 0 32 32"
              width={14}
            >
              <path
                className="recharts-legend-icon"
                d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                fill="none"
                stroke="#82ca9d"
                strokeWidth={4}
              />
            </svg>
            <span
              className="recharts-legend-item-text"
            >
              uv
            </span>
          </li>
        </ul>
      </div>
      <div
        className="recharts-tooltip-wrapper"
        style={
          Object {
            "MozTransform": "translate(undefinedpx, undefinedpx)",
            "OTransform": "translate(undefinedpx, undefinedpx)",
            "WebkitTransform": "translate(undefinedpx, undefinedpx)",
            "msTransform": "translate(undefinedpx, undefinedpx)",
            "pointerEvents": "none",
            "position": "absolute",
            "top": 0,
            "transform": "translate(undefinedpx, undefinedpx)",
            "visibility": "hidden",
          }
        }
      >
        <div
          className="recharts-default-tooltip"
          style={
            Object {
              "backgroundColor": "#fff",
              "border": "1px solid #ccc",
              "margin": 0,
              "padding": 10,
              "whiteSpace": "nowrap",
            }
          }
        >
          <p
            className="recharts-tooltip-label"
            style={
              Object {
                "margin": 0,
              }
            }
          >
            
          </p>
        </div>
      </div>
    </div>
  </div>
  <h2>
    Example Case:
  </h2>
  <p>
    Suppose that the Austrian government has heard about problematic forms of feeding ("fattening") geese in Member State Germany. Such geese are forcefully fed with excessive amounts of feedstuffs, which leads to the multiplication of their weight until the point of time of their slaughter. Suppose furthermore that such forms of feeding are prohibited in Austria and that no relevant EU secondary law exists which regulates such feeding methods. The Austrian government considers, against this background to implement an import ban for any geese from Germany, since it cannot be ruled out with certainty that the consumption of such fattened geese is dangerous to human health, even though such risk is highly unlikely. Will such a regulatory project comply with the fundamental freedoms of EU law?
  </p>
</div>
`;
